### ISSUE TITLE
Fix MNE connectivity feature calculation

### ISSUE THREAD

**Issue by toni-neurosc** [neuromodulation/py_neuromodulation#336]
MNE connectivity is super slow and I think it's because of a bug in their code. We might have to report to them. The call to `raw.get_data()` is taking more than 90% of the computation time of `get_epoched_data`, and it's only called to get the array shape... this is some kind of bug that we need to get rid of.

Also, we should disable MNE logger, it outputs to the console a lot. At the very least we should have a parameter that controls it, but by default have it off.

Btw, they're using a decorator inside their `mne.utils._logging.py` module that can be added to function to automatically set the logging level for functions, instead of using `if` statemetns. Could be sth we can learn from, idk

**Comment by timonmerk** [comment]
@toni-neurosc Can you print the profiling picture you're sharing earlier? And point out how it could be improved. Thomas from our lab is a core maintainer of the package. @ tsbinns

**Comment by toni-neurosc** [comment]
Hi @timonmerk , I don't think I can fulfill that request, the reason of the slow performance for `raw.get_data` in our code is not necessarily that its code is sub-optimal, is more than the function does a lot more than we required from it. In fact, is an almost ~200 lines of code mastodon of a function with a lot validations, type conversions, data copying... I can't evaluate whether all of that activity is or not required without knowing more about the required specifications from that function. 

What I know for sure is that we don't need it just to get the shape of the data we just passed to `RawArray` as we already had that information before even initializing the `RawArray` so we can live without `get_data`. That's all I can really say about it :/. And also in light of this, I don't think the profiling data is very relevant, the point of it was that it informed me that just getting the shape of the data was taking too long. 

Besides that this has been solved with #329 merge

**Comment by toni-neurosc** [comment]
Hi @timonmerk , I don't think I can fulfill that request, the reason of the slow performance for `raw.get_data` in our code is not necessarily that its code is sub-optimal, is more than the function does a lot more than we required from it. In fact, is an almost ~200 lines of code mastodon of a function with a lot validations, type conversions, data copying... I can't evaluate whether all of that activity is or not required without knowing more about the required specifications from that function. 

What I know for sure is that we don't need it just to get the shape of the data we just passed to `RawArray` as we already had that information before even initializing the `RawArray` so we can live without `get_data`. That's all I can really say about it :/. And also in light of this, I don't think the profiling data is very relevant, the point of it was that it informed me that just getting the shape of the data was taking too long. 

Besides that this has been solved with #329 merge
